src/modules/commander/HealthAndArmingChecks/checks/estimatorCheck.cpp

```diff
@@ -765,6 +766,44 @@ void EstimatorChecks::setModeRequirementFlags(const Context &context, bool pre_f
 		!checkPosVelValidity(now, xy_valid, gpos.eph, _param_com_pos_fs_eph.get(), gpos.timestamp,
 				     _last_gpos_fail_time_us, !failsafe_flags.global_position_invalid);
 
+	// Additional warning if the system is about to enter position-loss failsafe after dead-reckoning period
+	const float eph_critical = 2.5f * _param_com_pos_fs_eph.get(); // threshold used to trigger the navigation failsafe
+	const float gpos_critical_warning_thrld = math::max(0.9f * eph_critical, math::max(eph_critical - 10.f, 0.f));
+
+	estimator_status_flags_s estimator_status_flags;
+
+	if (_estimator_status_flags_sub.copy(&estimator_status_flags)) {
+		const bool dead_reckoning = estimator_status_flags.cs_inertial_dead_reckoning
+					    || estimator_status_flags.cs_wind_dead_reckoning;
+
+		if (!failsafe_flags.global_position_invalid
+		    && !failsafe_flags.nav_failure_imminent_warned
+		    && gpos.eph > gpos_critical_warning_thrld
+		    && dead_reckoning) {
+			/* EVENT
+			 * @description
+			 * Estimated position error is approaching the failsafe threshold.
+			 *
+			 * <profile name="dev">
+			 * This warning is triggered when the position error estimate is 90% of (or only 10m below) <param>COM_POS_FS_EPH</param> parameter.
+			 * </profile>
+			 */
+			events::send(events::ID("check_estimator_position_failure_imminent"), {events::Log::Error, events::LogInternal::Info},
+				     "Critical navigation failure, failsafe imminent. Switch to manual mode recommended");
```

Comment:
`Switch to manual mode recommended` could be part of the description.
And I would find it almost more helpful if the main message was `Estimated position error is approaching the failsafe threshold` instead of in the description.

Reply:
True, sounds better (though maybe a bit technical for some operators). d394b50a62ba0b8382f046d6ea7b24bcd791a422